WordPress/Classes/Services/MenusService.m

```diff
@@ -85,83 +85,39 @@ - (void)syncMenusForBlog:(Blog *)blog
 
 #pragma mark - Creating and updating menus
 
-- (void)createMenuWithName:(NSString *)menuName
-                      blog:(Blog *)blog
-                   success:(nullable MenusServiceCreateMenuRequestSuccessBlock)success
+- (void)createOrUpdateMenu:(Menu *)menu
+                   forBlog:(Blog *)blog
+                   success:(nullable MenusServiceCreateOrUpdateMenuRequestSuccessBlock)success
                    failure:(nullable MenusServiceFailureBlock)failure
 {
     NSParameterAssert([blog isKindOfClass:[Blog class]]);
-    NSAssert([self blogSupportsMenusCustomization:blog], @"Do not call this method on unsupported blogs, check with blogSupportsMenusCustomization first.");
-    
-    if (blog.wordPressComRestApi == nil) {
-        return;
-    }
-
-    MenusServiceRemote *remote = [[MenusServiceRemote alloc] initWithWordPressComRestApi:blog.wordPressComRestApi];
-    [remote createMenuWithName:menuName
-                          blog:blog
-                       success:^(RemoteMenu * _Nonnull remoteMenu) {
-                           [self.managedObjectContext performBlock:^{
-                               if (success) {
-                                   success(remoteMenu.menuID);
-                               }
-                           }];
-                       }
-                       failure:failure];
-}
-
-- (void)updateMenu:(Menu *)menu
-           forBlog:(Blog *)blog
-           success:(nullable MenusServiceUpdateMenuRequestSuccessBlock)success
-           failure:(nullable MenusServiceFailureBlock)failure
-{
-    NSParameterAssert([blog isKindOfClass:[Blog class]]);
     NSParameterAssert([menu isKindOfClass:[Menu class]]);
     NSAssert([self blogSupportsMenusCustomization:blog], @"Do not call this method on unsupported blogs, check with blogSupportsMenusCustomization first.");
     
     if (blog.wordPressComRestApi == nil) {
-        return;
+        return;   
```

Comment:
It is a bit undefined yeah. I added all those checks because before that, we'd try to create a remote with a nil api, which would often crash. The solution wasn't perfect but it was meant to prevent crashing.

I'm not sure if we should call `failure` or not. If we don't, some caller might be expecting either success/failure to update the UI. If we do, we have to create an error, and I can't think of a way you can add an error message that you can show to users and makes sense to them. "API object was nil" is definitely not helpful to users.

We can add an assertion to catch these cases on debug builds, but still doesn't solve the problem in production.